tick/survival/sccs/batch_convolution_sccs.py

```diff
@@ -0,0 +1,183 @@
+                solvers[-1].step = self.step
+                solvers[-1].set_model(model_list[j1]).set_prox(proxes[-1])
+                solvers[-1]._solver.set_epoch_size(model_list[j1]._model.get_epoch_size())
+        # no point calling "set_start_iterate" as it is always 0s
+        coeffes = solvers[0].multi_solve(coeffs_list, solvers, self.max_iter, None, set_start=False)
+        for ji in range(n_folds * len(C_s)):
+            coeffs_list[ji] = coeffes[ji]
+        return coeffes
+
+    def fit_kfold_cv(self, features, labels, censoring, C_tv_range: tuple = (),
+                         C_group_l1_range: tuple = (), logscale=True,
+                         n_cv_iter: int = 30, n_folds: int = 3,
+                         shuffle: bool = True, confidence_intervals: bool = False,
+                         n_samples_bootstrap: int = 100, confidence_level: float = .95):
+        p_features, p_labels, p_censoring = self._prefit(
+            features, labels, censoring)
+        # split the data with stratified KFold
+        kf = StratifiedKFold(n_folds, shuffle, self.random_state)
+        labels_interval = np.nonzero(p_labels)[1]
+
+        # Training loop
+        model_global_parameters = {
+            "n_intervals": self.n_intervals,
+            "n_lags": self.n_lags,
+            "n_features": self.n_features,
+        }
+        cv_tracker = CrossValidationTracker(model_global_parameters)
+        C_tv_generator, C_group_l1_generator = self._construct_generator_obj(
+            C_tv_range, C_group_l1_range, logscale)
+        batch_size = self.batch_size * n_folds
+        if batch_size > (n_cv_iter * n_folds):
+            batch_size = (n_cv_iter * n_folds)
+        model_list, coeffs_list  = ([] for i in range(2)) # 2 on the left
+        for i in range(batch_size):
+            model_list.append(self._construct_model_obj())
+            coeffs_list.append(np.zeros(self.n_coeffs))
+        j_cv_iter = (n_cv_iter // batch_size) * n_folds
+        t_cv_iter = batch_size // n_folds
+        i = 0
+        while i < j_cv_iter:
+            C_s, score_vals, indexes, train_scores, test_scores = ([] for i in range(5)) # 3 on the left
+
+            def multi_fit_job_setup(t_cv, n_job):
+                j1 = n_job + (t_cv * n_job)
+                train_index, test_index = indexes[-1][n_job]
+                train = itemgetter(*train_index.tolist())
+                test = itemgetter(*test_index.tolist())
+                X_train, X_test = list(train(p_features)), list(test(p_features))
+                y_train, y_test = list(train(p_labels)), list(test(p_labels))
+                censoring_train, censoring_test = p_censoring[train_index], \
+                    p_censoring[test_index]
+                model_list[j1].fit(X_train, y_train, censoring_train)
+                score_vals.append((X_test, y_test, censoring_test))
+
+            def run_multi_fit_jobs():
```

Comment:
Can you avoid this closure by passing explicitly these paramaters (model_list, coeffs_list, C_s, n_folds)?

Reply:
Would it be possibly to move this function outside?

Reply:
It would be possible but I fail to see the point, it's not likely to be used by anything but this function
the only reason for the closure is to avoid excessive indentation

Reply:
closures removed and just inlined